src/Server.hs

```diff
@@ -1,51 +1,61 @@
 module Server where
 
 import Weft.Internal.Types
+import Weft.Types
 import Weft.Generics.QueryParser
+import Weft.Generics.Resolve
+import Weft.Generics.EmptyQuery
 import TestData
 import Lens.Micro
 import Lens.Micro.Aeson
-import Network.Wai (responseLBS, Application, requestBody)
+import Network.Wai (responseLBS, Application, Response, getRequestBodyChunk)
 import Network.Wai.Handler.Warp (run)
-import Network.HTTP.Types (status200)
+import Network.HTTP.Types (status200, status500)
 import Network.HTTP.Types.Header (hContentType)
 import Data.Aeson hiding (json)
 import Data.Attoparsec.ByteString.Char8
 import Data.ByteString.Char8
-import Data.Text
 import Data.Text.Encoding
+import qualified Data.Map as M
 import qualified Data.ByteString.Lazy as BL
+import Control.Monad.Reader
 
-
-
-
-parseReqBody :: Text -> Either String (User 'Query)
+parseReqBody :: (HasEmptyQuery GqlQuery, HasQueryParser GqlQuery)
```

Comment:
@isovector eventually we have to let the user pass in their own GqlQuery type, as well as their GqlMutation and GqlSubscription type

Reply:
Try parsing this as a `Gql query` instead of a `GqlQuery`.

Reply:
Solved this kinda. Now we have to give the user a way to pass in their Gql type and the accompanying resolver